### PR TITLE
Do not create a loop that generates thousands of jobs

### LINKED ISSUE #21980: Do not load all cron jobs in one query and create all those isntances
> cron.php runs for more than 90min and the "MySQL server gone away" is not a rare error, it occurs on every call to cron.php.

@felixboehm

### PR DESCRIPTION
**Pull request by nickvergessen**
@PVince81 @MorrisJobke @DeepDiver1975 

Fix #21980

@karlitschek we should backport the cron.php commit to 8.2 at least

### CODE REVIEW

**Comment by karlitschek** [comment]
Awesome. :+1:  Please backport

File: lib/public/backgroundjob/ijoblist.php [hunk 1]
```diff
@@ -64,6 +64,8 @@ public function has($job, $argument);
 	 *
 	 * @return \OCP\BackgroundJob\IJob[]
 	 * @since 7.0.0
+	 * @deprecated 9.0.0 - This method is dangerous since it can cause load and
```
Comment (DeepDiver1975) [inline lib/public/backgroundjob/ijoblist.php]: yes - very important - THX

**Comment by DeepDiver1975** [comment]
Is getAll used in other places?

**Comment by nickvergessen** [comment]
Yes, looking at them atm

**Comment by MorrisJobke** [comment]
```

1) Test\Server::testQuery with data set #92 ('TagManager', '\OCP\ITagManager')
Doctrine\DBAL\DBALException: Failed to connect to the database: An exception occured in driver: SQLSTATE[HY000] [1040] Too many connections

/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/db/connection.php:53
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:429
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:389
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:621
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/db/connection.php:136
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DriverManager.php:172
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/db/connectionfactory.php:118
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/server.php:367
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/3rdparty/pimple/pimple/src/Pimple/Container.php:113
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/appframework/utility/simplecontainer.php:102
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/servercontainer.php:87
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/server.php:880
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/server.php:146
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/3rdparty/pimple/pimple/src/Pimple/Container.php:113
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/appframework/utility/simplecontainer.php:102
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/servercontainer.php:87
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/server.php:149
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/3rdparty/pimple/pimple/src/Pimple/Container.php:113
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/appframework/utility/simplecontainer.php:102
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/lib/private/servercontainer.php:87
/ssd/jenkins/workspace/core-ci-linux@3/database/mysql/label/SLAVE/tests/lib/server.php:173
```

Woot?

**Comment by MorrisJobke** [comment]
Tested and still works :+1: 

Also the code makes sense :)

**Comment by nickvergessen** [comment]
Okay, the remaining methods are deprecated since ownCloud6 and also reference a class which I couldn't find all the way through. Code was added by https://github.com/owncloud/core/pull/3051

I tested the methods and they still "work", but always return an empty array. If we really want to remove the getAll() usage from them, I'd go on and replace the content with `return [];` :speak_no_evil: 

But up to you @DeepDiver1975

**Comment by DeepDiver1975** [comment]
> I'd go on and replace the content with return [];

yes - since these classes no longer exist: kil with :fire:

**Comment by rullzer** [comment]
mysql tests are not happy.

**Comment by rullzer** [comment]
Locally everything still wotks!
Code looks good. We should really ban getAll type of functionlity.

:+1:

**Comment by nickvergessen** [comment]
Now mysql passed, maybe temporary problem, or unrelated? We don't really test cron.php so I would not know how this patch could have caused it.

**Pull request merged**